modules/l4remoteiplist/matcher.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2024 SICK AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l4remoteiplist
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+
+	"github.com/mholt/caddy-l4/layer4"
+	"go.uber.org/zap"
+)
+
+func init() {
+	caddy.RegisterModule(&RemoteIpList{})
+}
+
+type RemoteIpList struct {
+	RemoteIpFile string `json:"ip_file"`
+
+	logger       *zap.Logger
+	remoteIpList *IpList
+}
+
+// CaddyModule returns the Caddy module information.
+func (*RemoteIpList) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "layer4.matchers.remote_ip_list",
+		New: func() caddy.Module { return new(RemoteIpList) },
+	}
+}
+
+// Provision implements caddy.Provisioner.
+func (m *RemoteIpList) Provision(ctx caddy.Context) error {
+	m.logger = ctx.Logger()
+
+	remoteIpList, err := NewIpList(m.RemoteIpFile, &ctx, m.logger)
+	if err != nil {
+		m.logger.Error("error creating a new IP list", zap.Error(err))
+		return err
+	}
+	m.remoteIpList = remoteIpList
+	m.remoteIpList.StartMonitoring()
+	return nil
+}
+
+// The Match will return true if the remote IP is found in the remote IP list
+func (m *RemoteIpList) Match(cx *layer4.Connection) (bool, error) {
+	clientIP, err := m.getRemoteIP(cx)
+	if err != nil {
+		// Error, tread IP as matched
+		m.logger.Error("error parsing the remote IP from the connection", zap.Error(err))
+		return true, err
+	}
+
+	strClientIP := clientIP.String()
```

Comment:
If we match IP addresses, no need to convert clientIP into string. Besides, `clientIP` is a bit misguiding here for me, since Caddy http app has a `client_ip` matcher, that [differs](https://github.com/caddyserver/caddy/blob/master/modules/caddyhttp/ip_matchers.go) from `remote_ip` matcher. I suggest we stick to `remoteIP` to avoid confusion.